distributed/scheduler.py

```diff
@@ -3159,8 +3159,9 @@ def reschedule(self, key=None, worker=None):
     # Assigning Tasks to Workers #
     ##############################
 
-    def check_idle_saturated(self, worker):
-        occ = self.occupancy[worker]
+    def check_idle_saturated(self, worker, occ=0):
```

Comment:
I recommend `occ=None` here and `if occ is not None:`  Otherwise the meaning of `occ=0` is ambiguous between "there is no occupancy" and "please use the default behavior"

Reply:
I'm slightly concerned that other operations that call `check_idle_saturated` will now use the incorrect value in `Scheduler.occupancy` if they are called between requesting and confirming a steal.

Reply:
Agreed on occ=None.

If I understand the code correctly, as written now, the task is effectively owned by the victim until the theft is confirmed; before ownership was sort of blurred between request and confirmation. So, what bad things might happen with other things looking at check_idle_saturated? The only thing I could think of was that the thief gets killed - and move_task_confirm already tests for this.

Reply:
It might be placed back into idle or saturated, which would trigger additional work stealing.  This may not be that bad though.